crates/milli/src/vector/settings.rs

```diff
@@ -333,6 +739,223 @@ impl SettingsDiff {
+                    &mut headers,
+                    new_source,
+                    new_model,
+                    new_revision,
+                    new_pooling,
+                    new_api_key,
+                    new_dimensions,
+                    new_document_template,
+                    new_document_template_max_bytes,
+                    new_url,
+                    new_request,
+                    new_response,
+                    new_headers,
+                );
+
+                // update phony settings, it is always an error to have them set.
+                distribution.apply(new_distribution);
+                binary_quantized.apply(new_binary_quantized);
+                search_embedder.apply(new_search_embedder);
+                indexing_embedder.apply(new_indexing_embedder);
+
+                let updated_settings = SubEmbeddingSettings {
+                    source,
+                    model,
+                    revision,
+                    pooling,
+                    api_key,
+                    dimensions,
+                    document_template,
+                    url,
+                    request,
+                    response,
+                    headers,
+                    document_template_max_bytes,
+                    distribution,
+                    binary_quantized,
+                    search_embedder,
+                    indexing_embedder,
+                };
+                Setting::Set(updated_settings)
+            }
+            // handled during validation of the settings
+            Setting::Reset | Setting::NotSet => sub_embedder,
+        };
+        Ok(ret)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn diff(
+        reindex_action: &mut Option<ReindexAction>,
+        source: &mut Setting<EmbedderSource>,
+        model: &mut Setting<String>,
+        revision: &mut Setting<String>,
+        pooling: &mut Setting<OverridePooling>,
+        api_key: &mut Setting<String>,
+        dimensions: &mut Setting<usize>,
+        document_template: &mut Setting<String>,
+        document_template_max_bytes: &mut Setting<usize>,
+        url: &mut Setting<String>,
+        request: &mut Setting<serde_json::Value>,
+        response: &mut Setting<serde_json::Value>,
+        headers: &mut Setting<BTreeMap<String, String>>,
+        new_source: Setting<EmbedderSource>,
+        new_model: Setting<String>,
+        new_revision: Setting<String>,
+        new_pooling: Setting<OverridePooling>,
+        new_api_key: Setting<String>,
+        new_dimensions: Setting<usize>,
+        new_document_template: Setting<String>,
+        new_document_template_max_bytes: Setting<usize>,
+        new_url: Setting<String>,
+        new_request: Setting<serde_json::Value>,
+        new_response: Setting<serde_json::Value>,
+        new_headers: Setting<BTreeMap<String, String>>,
+    ) {
+        // **Warning**: do not use short-circuiting || here, we want all these operations applied
```

Comment:
I was a bit surprised to see that the `diff` function was actually mostly modifying its parameter, would be nice to find a better name, but I don’t have one 😅

Reply:
`apply_and_diff`?